### PR TITLE
i1690: Add log viewing capabilities, and log each run to separate file

### PR DESCRIPTION
**Pull request by MartinEden**
https://vimc.myjetbrains.com/youtrack/issue/VIMC-1690

With this change you can run `bb8 log` to view the most recent log, no muss, no fuss. There are also options to list available log files, and to view a particular one.

Rich, I decided I was overthinking this: If you want to use `less`, you can just run `bb8 log | less`. This might take a few seconds if its a big log file while the whole thing travels via the pipe, as opposed to less reading as it needs to, but I think that's acceptable.

### CODE REVIEW

**Review by richfitz — COMMENTED** [review]
This looks good to me but I have not run it - is there a simple way of testing things (e.g., running this against the  bb8 installations that are running at the moment?)

**Comment by MartinEden** [comment]
I just tested it locally. Easy enough:

```
sudo ./setup ../config.json orderly
bb8 restore # Repeat this to get more logs
bb8 log --list
etc.
```

**Review by hillalex — APPROVED** [review]

File: bin/main.py [hunk 1]
```diff
@@ -5,34 +5,48 @@
   main.py backup
   main.py restore
   main.py init
+  main.py log --list [--limit=<number>]
+  main.py log [--version=<version>]
```
Comment (richfitz) [inline bin/main.py]: Would this second form would be more idiomatic as `main.py log [<version>]`?

Comment (richfitz) [reply bin/main.py]: Or even as `main.py log list` or `main.py log show [<version>]` as the two forms

Comment (richfitz) [reply bin/main.py]: Also, the name `main.py` should probably be bb8 so that running `bb8 --help` gives more understandable output

**Review by richfitz — CHANGES_REQUESTED** [review]
Minor changes here, but confirmed working and nice to use! I suspect that a fully subcommanded approach would be best - I think we're going to end up eventually wanting something like `bb8 log grep <pattern>` and that would allow that unambiguously

File: bin/main.py (hunk 1 above)
Comment (MartinEden) [reply bin/main.py]: How's this?

Comment (richfitz) [reply bin/main.py]: nice!

**Review by richfitz — APPROVED** [review]

**Pull request merged**